Dockerfile

```diff
@@ -1,39 +1,23 @@
-# Taken from https://github.com/docker-library/golang/blob/master/1.6/Dockerfile 
-# to build our golang image with debian testing (stretch).
-FROM buildpack-deps:stretch-scm
-
-# gcc for cgo.
-RUN apt-get update && apt-get install -y --no-install-recommends \
-		g++ \
-		gcc \
-		libc6-dev \
-		make \
-	&& rm -rf /var/lib/apt/lists/*
-
-ENV GOLANG_VERSION 1.7.1
-ENV GOLANG_DOWNLOAD_URL https://golang.org/dl/go$GOLANG_VERSION.linux-amd64.tar.gz
-ENV GOLANG_DOWNLOAD_SHA256 43ad621c9b014cde8db17393dc108378d37bc853aa351a6c74bf6432c1bbd182
-
-RUN curl -fsSL "$GOLANG_DOWNLOAD_URL" -o golang.tar.gz \
-	&& echo "$GOLANG_DOWNLOAD_SHA256  golang.tar.gz" | sha256sum -c - \
-	&& tar -C /usr/local -xzf golang.tar.gz \
-	&& rm golang.tar.gz
-
-ENV GOPATH /go
-ENV PATH $GOPATH/bin:/usr/local/go/bin:$PATH
-
-RUN mkdir -p "$GOPATH/src" "$GOPATH/bin" && chmod -R 777 "$GOPATH"
-WORKDIR $GOPATH
+FROM golang:1.8
```

Comment:
Does the above change break builds for x86 platforms? Specifically, does the above change break the Intel RDT/CAT library build dependencies?

Reply:
Hi @jaypipes 
The content of this Dockerfile is too redundant.
To make it clean, I used the multi-arch supported image to replace it.
golang:1.8 support multi-arch.
And also it not break the Intel RDT/CAT library build dependencies. 
I have tested it.  No problems were found in x86 platform.

Reply:
OK, thanks for checking on that @lubinsz. I still think that you should separate out the "get multi-platform Dockerfile working" changes from the "introduce support for NFD" changes, though. Can you do that?

Reply:
@lubinsz The content you deleted was added to reduce vulnerabilities in the Docker image. See https://github.com/kubernetes-incubator/node-feature-discovery/pull/13.

Reply:
@balajismaniam I see the code from issue #13, but I actually see no indication of what specific vulnerabilities in the Docker image were addressed by those changes. I think it's probably good to remove those lines and go with the straight FROM golang:1.8 base unless there's an important reason to diverge.

Reply:
@balajismaniam 
I see. 
I have updated the code by following  #13 
Currently, starting from golang version 1.85,  there is an official binary package for arm64 available for download.
So the golang version should be updated into 1.8.5 at least.
Please see the code as more reference.

Reply:
@jaypipes I added the comment just for reference. You see the differences in the vulnerabilities in quay.io. There were some packages in golang base image, which introduced some vulnerabilities. I'm fine with using just `FROM golang:1.8`. 
@lubinsz Please revert back to using `FROM golang:1.8`. Sorry for the confusion.